### DISCUSSION TITLE
Mismatch between provided laion400 text embedding .npy files and the clip ViT-B/32 text encoder output

### DISCUSSION THREAD

**Discussion by TianRui-Song717** [rom1504/clip-retrieval#271] [Q&A]
When processing LAION400 embedding files, I found that the embedding in the provided files: https://the-eye.eu/public/AI/cah/laion400m-met-release/laion400m-embeddings/texts/text_emb_0.npy
Does not match the output of clip text encoder. 
```python
clip_encoder, _ = clip.load("ViT-B/32", device='cpu')
caption = pd.read_parquet("H:\\LAION\\embeddings\\metadata\\metadata_0.parquet").loc[0, "caption"]

tokenize_caption = clip.tokenize([caption])
with torch.no_grad():
    caption_emb = clip_encoder.encode_text(tokenize_caption)
    caption_emb = caption_emb.detach().numpy()

laion_text_emb = np.load("H:\\LAION\\embeddings\\texts\\text_emb_0.npy")[0]
```
Here the `laion_text_emb` does not match `caption_emb `, even when I normalize the clip output `caption_emb / np.linalg.norm(caption_emb)`, they are still not the same, what's the problem?

**Comment by rom1504** [comment]
See https://github.com/rom1504/clip-retrieval/issues/264
